### ISSUE TITLE
[Feature Request] callback afterFind

### ISSUE THREAD

**Issue by jurajpetrik** [balderdashy/waterline#923]
Would it be possible to implement a  callback afterFind? E.g. I have an event with a date specified and in afterFind I could insert an attribute hasExpired..

**Comment by dmarcelino** [comment]
This has also been discussed in #525 ([comment](https://github.com/balderdashy/waterline/pull/525#issuecomment-86678830)).

@aclave1, would you be interested in taking this?

I think this is a good fit for `0.11`.

cc: @randallmeeker

**Comment by randallmeeker** [comment]
+1 for v0.11

**Comment by anasyb** [comment]
One minute I think this should go into sails another min I think it should go into waterline. But definitely it should go somewhere!!
I was experimenting with **Model Policies** (different route policies that sails provide us with) which would also have access to the `req` as well as the returned `records`.. and run between the model action and the controller. @jurajpetrik checkout the discussion and gist here https://github.com/balderdashy/sails/issues/2760
Its a simple hook you drop into api/hooks directory... would love to hear your feedback! Gist here https://gist.github.com/anasyb/c2197a0d47ef29ee7418

**Comment by anasyb** [comment]
@jurajpetrik for your particular use case have you considered an attribute function, something along the lines of:
```javascript
//in some model
attributes: {
  date:...
  //...
  hasExpired: function (){ return this.date < new Date();}
}
```

Or is it that you want it to be sent over `res` or persisted?

**Comment by aclave1** [comment]
@dmarcelino I can take this. When should I do this?

**Comment by dmarcelino** [comment]
How about 0.11? There is no scheduled date yet... And thanks!

**Comment by jurajpetrik** [comment]
@anasyb Thanks I didn't know about attribute functions, I do want it to be sent over res though.

**Comment by anasyb** [comment]
Then maybe you have two options:
option 1.  move that to the controller or the blueprint, or even a policy (depending on your case). You can keep the attr function if it has some use elsewhere.
```javascript
//api/controllers/EventController.js
module.exports = {
  findOne: function( req, res ) {
    var eventId;// = bla bla.. get it from req or from wherever you're getting it!
    Event.findOne(eventId).exec(function(err, eve){
      if ( err ) return res.serverError( err );
      if ( !eve ) return res.notFound( 'No event found with ID ' + eventId );

      var eveObj = eve.toObject();
      //temp workaround for https://github.com/balderdashy/waterline/issues/891#issuecomment-88361748
      eveObj.toJSON = undefined;
      eveObj.hasExpired = eve.hasExpired();//assuming you've implemented hasExpired as an attr func

      res.ok(eveObj);//bye bye darling
    });
  }
};
```

option 2. override toJSON in your model to add all that fancy stuff. This is a good option if you always want to add that attr to the json of your event. You can find an example of overriding `toJSON` in this comment https://github.com/balderdashy/waterline/issues/317#issuecomment-38376374

Good luck!

**Comment by mikermcneil** [comment]
Thanks for posting, @jurajpetrik.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/waterline/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by devinivy** [comment]
Reopening since this is a feature request in a current milestone.

**Comment by LunaPg** [comment]
+1 :+1:

**Comment by sailsbot** [comment]
Thanks for posting, @jurajpetrik.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by tyronedougherty** [comment]
Hi all, is there any update on the status of this feature? An afterFind callback would be really useful for me.

Thanks!

**Comment by Seafnox** [comment]
An afterFind lifecycle callback would be really useful for me too!

**Comment by hitokun-s** [comment]
+1

**Comment by Bikappa** [comment]
+1
